satellite/satellitedb/overlaycache.go

```diff
@@ -930,49 +931,61 @@ func (cache *overlaycache) AllPieceCounts(ctx context.Context) (_ map[storj.Node
 }
 
 func (cache *overlaycache) UpdatePieceCounts(ctx context.Context, pieceCounts map[storj.NodeID]int) (err error) {
-	// TODO: add monkit stuff
 	defer mon.Task()(&ctx)(&err)
-
 	if len(pieceCounts) == 0 {
 		return nil
 	}
 
-	switch cache.db.Driver().(type) {
-	case *pq.Driver:
-		return Error.Wrap(cache.postgresUpdatePieceCounts(ctx, pieceCounts))
-	default:
-		return Error.Wrap(cache.sqliteUpdatePieceCounts(ctx, pieceCounts))
+	type NodeCount struct {
+		ID    storj.NodeID
+		Count int64
 	}
-}
+	var counts []NodeCount
 
-func (cache *overlaycache) postgresUpdatePieceCounts(ctx context.Context, pieceCounts map[storj.NodeID]int) (err error) {
-	defer mon.Task()(&ctx)(&err)
+	for nodeid, count := range pieceCounts {
+		counts = append(counts, NodeCount{
+			ID:    nodeid,
+			Count: int64(count),
+		})
+	}
 
-	sqlQuery := `UPDATE nodes SET piece_count = newvals.piece_count FROM ( VALUES `
-	args := make([]interface{}, 0, len(pieceCounts)*2)
-	for nodeID, pieceCount := range pieceCounts {
-		sqlQuery += `(?::BYTEA, ?::BIGINT), `
-		args = append(args, nodeID, pieceCount)
-	}
-	sqlQuery = sqlQuery[:len(sqlQuery)-2]
-	// trim off the last comma+space
-	sqlQuery += `) newvals(nodeid, piece_count) WHERE nodes.id = newvals.nodeid`
-	_, err = cache.db.DB.ExecContext(ctx, cache.db.Rebind(sqlQuery), args...)
-	return err
-}
+	sort.Slice(counts, func(i, k int) bool {
```

Comment:
Just curious why not use `.SliceStable` here?

Reply:
Because the node ids are guaranteed to be unique, so there wouldn’t be a difference in the result.